<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- muscle is true for muscle hand or false for motor hand -->
  <!-- prefix can be anything but usually is "r_" or  "l_" for right and left hands distinction -->
  <xacro:macro name="forearm" params="muscle prefix">
    <link name="${prefix}forearm">
      <inertial>
        <origin xyz="0 0 0.09" rpy="0 0 0" />
        <mass value="3.0" />
        <inertia ixx="0.0138" ixy="0.0" ixz="0.0" iyy="0.0138"
        iyz="0.0" izz="0.00744" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry name="${prefix}forearm_visual">
          <xacro:if value="${muscle}">
            <mesh filename="package://sr_description/hand/model/forearm_muscle.dae"
            scale="1.0 1.0 1.0" />
          </xacro:if>
          <xacro:unless value="${muscle}">
            <mesh filename="package://sr_description/hand/model/forearm.dae"
            scale="0.001 0.001 0.001" />
          </xacro:unless>
        </geometry>
        <material name="Grey" />
      </visual>
      <collision>
        <xacro:if value="${muscle}">
          <origin xyz="0 0 0.185" rpy="0 0 0" />
          <geometry name="${prefix}forearm_collision">
            <cylinder radius="0.075" length="0.37" />
          </geometry>
        </xacro:if>
        <xacro:unless value="${muscle}">
          <origin xyz="0 0 0.092" rpy="0 0 0" />
          <geometry name="${prefix}forearm_collision">
            <box size="0.140 0.140 0.184" />
          </geometry>
        </xacro:unless>
      </collision>
      <!-- wrist mount -->
      <collision>
        <xacro:if value="${muscle}">
          <origin xyz="0 0 0.395" rpy="0 0 0" />
          <geometry>
            <box size="0.03 0.04 0.06" />
          </geometry>
        </xacro:if>
        <xacro:unless value="${muscle}">
          <origin xyz="0 -0.01 0.181" rpy="0 0.78 0" />
          <geometry>
            <box size="0.07 0.07 0.07" />
          </geometry>
        </xacro:unless>
      </collision>
    </link>
    <gazebo reference="${prefix}forearm">
      <selfCollide>false</selfCollide>
    </gazebo>
  </xacro:macro>
</robot>
